domain/modelconfig/state/state.go

```diff
@@ -107,25 +107,21 @@ func (st *State) ModelConfig(ctx context.Context) (map[string]string, error) {
 		return config, errors.Trace(err)
 	}
 
-	return config, db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		stmt := `SELECT key, value FROM model_config`
-		rows, err := tx.QueryContext(ctx, stmt)
-		if err != nil {
-			return fmt.Errorf("getting model config values: %w", err)
+	stmt, err := st.Prepare(`SELECT &dbKeyValue.* FROM model_config`, dbKeyValue{})
+	if err != nil {
+		return config, errors.Trace(err)
+	}
+
+	return config, db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
```

Comment:
This pattern here, mutating a return value whilst calculating another return value, is a little evil imo

While you're here, maybe worth splitting up a little?